src/main/java/com/uid2/shared/secret/SecureKeyGenerator.java

```diff
@@ -22,11 +28,57 @@ public String generateRandomKeyString(int keyLen) {
     }
 
     @Override
-    public String generateFormattedKeyString(int keyLen) {
+    public KeyGenerationResult generateFormattedKeyStringAndKeyHash(String keyPrefix, int keyLen) {

```

Comment:
There's too much going on here, the salting and hashing should take after the prefix has been prepended

Why does the hash need a prefix prepended?

Can we just feed the UTF-8 bytes of the full key string into the hash function?